To: vim_dev@googlegroups.com
Subject: Patch 7.3.064
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.064
Problem:    Win32: ":dis +" shows nothing, but "+p does insert text.
Solution:   Display the * register, since that's what will be inserted.
	    (Christian Brabandt)
Files:	    src/globals.h, src/ops.c


*** ../vim-7.3.063/src/globals.h	2010-08-15 21:57:27.000000000 +0200
--- src/globals.h	2010-11-24 14:06:47.000000000 +0100
***************
*** 510,515 ****
--- 510,516 ----
  EXTERN VimClipboard clip_plus;	/* CLIPBOARD selection in X11 */
  # else
  #  define clip_plus clip_star	/* there is only one clipboard */
+ #  define ONE_CLIPBOARD
  # endif
  EXTERN int	clip_unnamed INIT(= FALSE);
  EXTERN int	clip_autoselect INIT(= FALSE);
*** ../vim-7.3.063/src/ops.c	2010-10-09 17:21:42.000000000 +0200
--- src/ops.c	2010-11-24 14:26:25.000000000 +0100
***************
*** 3979,3985 ****
      for (i = -1; i < NUM_REGISTERS && !got_int; ++i)
      {
  	name = get_register_name(i);
! 	if (arg != NULL && vim_strchr(arg, name) == NULL)
  	    continue;	    /* did not ask for this register */
  
  #ifdef FEAT_CLIPBOARD
--- 3979,3990 ----
      for (i = -1; i < NUM_REGISTERS && !got_int; ++i)
      {
  	name = get_register_name(i);
! 	if (arg != NULL && vim_strchr(arg, name) == NULL
! #ifdef ONE_CLIPBOARD
! 	    /* Star register and plus register contain the same thing. */
! 		&& (name != '*' || vim_strchr(arg, '+') == NULL)
! #endif
! 		)
  	    continue;	    /* did not ask for this register */
  
  #ifdef FEAT_CLIPBOARD
*** ../vim-7.3.063/src/version.c	2010-11-24 12:35:14.000000000 +0100
--- src/version.c	2010-11-24 14:24:03.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     64,
  /**/

-- 
FIRST SOLDIER:  So they wouldn't be able to bring a coconut back anyway.
SECOND SOLDIER: Wait a minute! Suppose two swallows carried it together?
FIRST SOLDIER:  No, they'd have to have it on a line.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
